editor/svgutils.js

```diff
@@ -529,19 +529,387 @@ svgedit.utilities.getBBox = function(elem) {
 	return ret;
 };
 
-// Function: svgedit.utilities.getRotationAngle
-// Get the rotation angle of the given/selected DOM element
+svgedit.utilities.getPathDFromSegments = function( pathSegments) {
+	var d = '';
+
+	$.each(pathSegments, function(j, seg) {
+		var i;
+		var l = seg[0], pts = seg[1];
+		d += l;
+		for (i = 0; i < pts.length; i+=2) {
+			d += (pts[i] +','+pts[i+1]) + ' ';
```

Comment:
It's been awhile since I looked at SVG code, but this assumes that the segments are only two coordinates a piece, right?  Aren't there some segments that are multiple numbers (arcs and the like)